src/lib/rmt/wizard_firewall_page.rb

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) 2018 SUSE LLC.
+#  All Rights Reserved.
+
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of version 2 or 3 of the GNU General
+#  Public License as published by the Free Software Foundation.
+
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+#  GNU General Public License for more details.
+
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, contact SUSE LLC.
+
+#  To contact SUSE about this file by physical or electronic mail,
+#  you may find current contact information at www.suse.com
+
+require 'y2firewall/firewalld'
+require 'rmt/utils'
+require 'cwm/dialog'
+require 'cwm/custom_widget'
+
+Yast.import 'CWMFirewallInterfaces'
+
+module RMT; end
+
+class RMT::WizardFirewallPage < CWM::Dialog
+  def initialize
+    textdomain 'rmt'
+  end
+
+  def title
+    _('RMT configuration step 4/5')
+  end
+
+  def abort_button
+    Yast::Label.CancelButton # to be consistent with the other pages
+  end
+
+  def contents
+    if firewalld.installed? && firewalld.enabled?
+      HBox(
+        HStretch(),
+        FirewallWidget.new,
+        HStretch()
+      )
+    else
+      VCenter(
+        Label(_("Firewalld is not enabled.\n\nIf you enable firewalld later,\nremember to open the firewall ports for HTTP and HTTPS."))
```

Comment:
Would "Firewall is not enabled/installed"... gives a hint that the (unlikely) case that it's not even installed is the reason they are seeing this? Alternatively, separate the installed? and enabled? cases and have two different strings that explain each case verbosely.

Reply:
Well, when there is no firewalld installed, saying "not enabled" is still correct (enough). I already thought about it when writing these lines, and decided that simplicity of code in this case is more important than being overly smart in the output. OK with you to keep it as is? :)

Reply:
Sure.